File: 1403-minimum-subsequence-in-non-increasing-order/1403-minimum-subsequence-in-non-increasing-order.java
class Solution {
    public List<Integer> minSubsequence(int[] nums) {
        Arrays.sort(nums);
        int total = Arrays.stream(nums).sum();
        int curr = 0;
        int index = nums.length - 1;
        List<Integer> answer = new ArrayList<>();
        while (index >= 0) {
            if (curr <= total) {
                answer.add(nums[index]);
                curr += nums[index];
                total -= nums[index];
            }
            index--;
        }
        return answer;
    }
}

File: 1403-minimum-subsequence-in-non-increasing-order/1403-minimum-subsequence-in-non-increasing-order.kt
class Solution {
fun minSubsequence(ar: IntArray): List<Int> = with(ar.sortedDescending()) {
    for (i in ar.indices) {
        val l = subList(0, i + 1)
        if (l.sum() > subList(i + 1, lastIndex + 1).sum()) return l
    }
    return emptyList()
}
}